chia/wallet/wallet_state_manager.py

```diff
@@ -934,335 +937,363 @@ async def new_coin_state(
-                                    assert isinstance(wallet, DataLayerWallet)
-                                    dl_wallet = wallet
-                                    break
-                            else:  # No DL wallet exists yet
-                                dl_wallet = await DataLayerWallet.create_new_dl_wallet(
-                                    self,
-                                    self.main_wallet,
-                                )
-                            await dl_wallet.track_new_launcher_id(
-                                child.coin.name(),
-                                peer,
-                                spend=launcher_spend,
-                                height=uint32(child.spent_height),
-                            )
-                        continue
-
-                    # solution_to_pool_state may return None but this may not be an error
-                    if pool_state is None:
-                        self.log.debug("solution_to_pool_state returned None, ignore and continue")
-                        continue
-
-                    assert child.spent_height is not None
-                    pool_wallet = await PoolWallet.create(
-                        self,
-                        self.main_wallet,
-                        child.coin.name(),
-                        [launcher_spend],
-                        uint32(child.spent_height),
-                        name="pool_wallet",
-                    )
-                    launcher_spend_additions = launcher_spend.additions()
-                    assert len(launcher_spend_additions) == 1
-                    coin_added = launcher_spend_additions[0]
-                    coin_name = coin_added.name()
-                    existing = await self.coin_store.get_coin_record(coin_name)
-                    if existing is None:
-                        await self.coin_added(
-                            coin_added,
-                            uint32(coin_state.spent_height),
-                            [],
-                            pool_wallet.id(),
-                            WalletType(pool_wallet.type()),
-                            peer,
-                            coin_name,
-                        )
-                    await self.add_interested_coin_ids([coin_name])
-
-            else:
-                raise RuntimeError("All cases already handled")  # Logic error, all cases handled
+                    else:
+                        raise RuntimeError("All cases already handled")  # Logic error, all cases handled
+            except Exception as e:
+                self.log.exception(f"Error adding state... {e}")
+                if isinstance(e, PeerRequestException) or isinstance(e, aiosqlite.Error):
+                    await self.retry_store.add_state(coin_state, peer.peer_node_id, fork_height)
+                else:
+                    await self.retry_store.remove_state(coin_state)
```

Comment:
it's removing the retry state when it starts the `new_coin_state` update why remove it again?

Reply:
The first removal is in the transaction that gets rolled back if there's a raise.  We need to add another command to remove it here in case an exception is thrown but it's not one of the blessed ones.